src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -0,0 +1,779 @@
+  const [isDisabled, setIsDisabled] = useState(false);
+  const firstRef = useRef<HTMLButtonElement>(null);
+
+  return (
+    <ModalContext.Provider value={{ isAnimationComplete, triggerRefocusFlag }}>
+      <FocusTrap
+        wrapperRef={ref}
+        {...rest}
+        isOpen
+        focusFirstElement={shouldFocusFirstElement ? firstRef : undefined}
+      >
+        <div ref={ref} data-testid={WRAPPER_ID} tabIndex={tabIndex}>
+          {React.Children.map(children, (child) => {
+            const focusableChild = child as React.ReactElement;
+
+            if (focusableChild?.props?.id === "disable-on-focus") {
+              return React.cloneElement(focusableChild, {
+                onFocus: () => setIsDisabled(true),
+                disabled: isDisabled,
+              });
+            }
+
+            return child;
+          })}
+          {shouldFocusFirstElement && (
+            <button type="button" ref={firstRef}>
+              {FIRST_ELEMENT}
+            </button>
+          )}
+        </div>
+      </FocusTrap>
+    </ModalContext.Provider>
+  );
+};
+
+describe("FocusTrap", () => {
+  describe("triggerRefocusFlag", () => {
+    it("refocuses the last element that had focus within the trap when flag is set", () => {
+      const { rerender } = render(
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <Select
+            id="c0499f86-d5a7-4a72-a0b7-753a2d218c54"
+            label="the dropdown"
+            aria-label="input"
+          >
+            <Option value="1" text="Option 1" />
+          </Select>
+        </MockComponent>
+      );
+      focusElement(screen.getByLabelText("input"));
+      screen.getByLabelText("input").blur();
+      expect(screen.getByLabelText("input")).not.toBeFocused();
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag>
+          <button type="button">{BUTTON_ONE}</button>
+          <Select
+            id="c0499f86-d5a7-4a72-a0b7-753a2d218c54"
+            label="the dropdown"
+            aria-label="input"
+          >
+            <Option value="1" text="Option 1" />
+          </Select>
+        </MockComponent>
+      );
+      expect(screen.getByLabelText("input")).toBeFocused();
+    });
+
+    it("refocuses the wrapper element when flag is set, if the wrapper has a tabindex", () => {
+      const { rerender } = render(
+        <MockComponent
+          autoFocus={false}
+          triggerRefocusFlag={false}
+          tabIndex={-1}
+        >
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag tabIndex={-1}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
```

Comment:
```suggestion
      const { rerender } = render(
        <MockComponent
          autoFocus={false}
          triggerRefocusFlag={false}
          tabIndex={-1}
        >
          <button type="button">{BUTTON_ONE}</button>
          <input type="text" />
        </MockComponent>
      );
      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
      rerender(
        <MockComponent autoFocus={false} triggerRefocusFlag tabIndex={-1}>
          <button type="button">{BUTTON_ONE}</button>
          <input type="text" />
        </MockComponent>
      );
      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
```
Would we need to verify the wrapper had focus before the flag was set as well so we are testing it stays in focus?